cdap-api/src/main/java/co/cask/cdap/api/annotation/Requirements.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.api.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * <p>Annotates the requirements needed by a plugin to run successfully.</p>
+ *
+ * <p>Requirements are case insensitive.</p>
+ *
+ * <p>If a plugin does not specify any {@link Requirements} or the {@link #value()} array is empty then it assumed
+ * that the plugin does not have any specific requirements and can run everywhere.</p>
+ *
+ * <p>Usage Examples:</p>
+ * <ul>
+ * <li><b>Omitting Requirements:</b> A plugin can chose to not specify any requirement by not specifying a
+ * {@link Requirements} annotation. In this case the plugin will considered to be capable or run everywhere.</li>
+ * <pre>
+ *     {@literal @}Plugin(type = BatchSource.PLUGIN_TYPE)
```

Comment:
seems like the `@` is off by a single space. Or does it get rendered differently?

Reply:
It gets rendered differently. the spacing is to render it correctly. 
![image](https://user-images.githubusercontent.com/1472867/44240985-f0eb5400-a175-11e8-8d9b-dfaf820b76e9.png)